aspnetcore/fundamentals/app-state.md

```diff
@@ -227,33 +230,17 @@ Data can be saved in hidden form fields and posted back on the next request. Thi
 
 The [HttpContext.Items](/dotnet/api/microsoft.aspnetcore.http.httpcontext.items) collection is used to store data while processing a single request. The collection's contents are discarded after a request is processed. The `Items` collection is often used to allow components or middleware to communicate when they operate at different points in time during a request and have no direct way to pass parameters.
 
-In the following example, [middleware](xref:fundamentals/middleware/index) adds `isVerified` to the `Items` collection.
-
-```csharp
-app.Use(async (context, next) =>
-{
-    // perform some verification
-    context.Items["isVerified"] = true;
-    await next.Invoke();
-});
-```
-
-Later in the pipeline, another middleware can access the value of `isVerified`:
+In the following example, [middleware](xref:fundamentals/middleware/index) adds `isVerified` to the `Items` collection:
 
-```csharp
-app.Run(async (context) =>
-{
-    await context.Response.WriteAsync($"Verified: {context.Items["isVerified"]}");
-});
-```
+[!code-csharp[](app-state/samples/3.x/SessionSample/Startup4.cs?name=snippet1)]
 
-For middleware that's only used by a single app, `string` keys are acceptable. Middleware shared between app instances should use unique object keys to avoid key collisions. The following example shows how to use a unique object key defined in a middleware class:
+-For middleware that's only used by a single app, fixed `string` keys are acceptable. Middleware shared between app instances should use unique object keys to avoid key collisions. The following example shows how to use a unique object key defined in a middleware class:
```

Comment:
```suggestion
For middleware that's only used in a single app, fixed `string` keys are acceptable. Middleware shared between apps should use unique object keys to avoid key collisions. The following example shows how to use a unique object key defined in a middleware class:
```